metricbeat/module/prometheus/collector/collector.go

```diff
@@ -140,3 +162,58 @@ func (m *MetricSet) addUpEvent(eventList map[string]common.MapStr, up int) {
 	}
 
 }
+
+func (m *MetricSet) skipFamily(family *dto.MetricFamily) bool {
+	// example:
+	//	include_metrics:
+	//		- node_*
+	//	exclude_metrics:
+	//		- node_disk_*
+	//
+	// This would mean that we want to keep only the metrics that start with node_ prefix but
+	// are not related to disk so we exclude node_disk_* metrics from them.
+
+	if family == nil {
+		return true
+	}
+
+	// if include_metrics are defined, check if this metric should be included
+	if len(m.includeMetrics) > 0  {
+		if !matchMetricFamily(*family.Name, m.includeMetrics) {
+			return true
+		}
+	}
+	// now exclude the metric if it matches any of the given patterns
+	if len(m.ignoreMetrics) > 0 {
+		if matchMetricFamily(*family.Name, m.ignoreMetrics) {
+			return true
+		}
+	}
+	return false
+}
+
+func compilePatternList(patterns *[]string) ([]*regexp.Regexp, error){
+	var compiledPatterns []*regexp.Regexp
+	compiledPatterns = []*regexp.Regexp{}
+	if patterns != nil {
+		for _, pattern := range *patterns {
+			r, err := regexp.Compile(pattern)
+			if err != nil {
+				return nil, err
```

Comment:
Here we may also want to include the pattern as context for this error, in case it is not already included in the error.
```suggestion
				return nil, errors.Wrapf(err, "compiling pattern '%s'", pattern)
```

Reply:
👍